source/workflowapi/app.py

```diff
@@ -2230,6 +2414,505 @@ def update_workflow_execution_status(id, status, message):
+#     / \ \      / / ___|  / ___|  ___ _ ____   _(_) ___ ___  |  _ \ _ __ _____  _(_) ___ ___
+#    / _ \ \ /\ / /\___ \  \___ \ / _ | '__\ \ / | |/ __/ _ \ | |_) | '__/ _ \ \/ | |/ _ / __|
+#   / ___ \ V  V /  ___) |  ___) |  __| |   \ V /| | (_|  __/ |  __/| | | (_) >  <| |  __\__ \
+#  /_/   \_\_/\_/  |____/  |____/ \___|_|    \_/ |_|\___\___| |_|   |_|  \___/_/\_|_|\___|___/
+#
+# ================================================================================================
+
+@app.route('/service/transcribe/get_vocabulary', cors=True, methods=['POST'], content_types=['application/json'], authorizer=authorizer)
+def get_vocabulary():
+    """ Get the description for an Amazon Transcribe custom vocabulary.
+
+    Returns:
+        This is a proxy for boto3 get_vocabulary and returns the output from that SDK method.
+        See `the boto3 documentation for details <https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/transcribe.html#TranscribeService.Client.get_vocabulary>`_
+
+    Raises:
+        See `the boto3 documentation for details <https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/transcribe.html#TranscribeService.Client.get_vocabulary>`_
+    """
+    print('get_vocabulary request: '+app.current_request.raw_body.decode())
+    transcribe_client = boto3.client('transcribe', region_name=os.environ['AWS_REGION'])
+    vocabulary_name = json.loads(app.current_request.raw_body.decode())['vocabulary_name']
+    response = transcribe_client.get_vocabulary(VocabularyName=vocabulary_name)
+    # Convert time field to a format that is JSON serializable
+    response['LastModifiedTime'] = response['LastModifiedTime'].isoformat()
+    return response
+
+
+@app.route('/service/transcribe/download_vocabulary', cors=True, methods=['POST'], content_types=['application/json'], authorizer=authorizer)
+def download_vocabulary():
+    """ Get the contents of an Amazon Transcribe custom vocabulary.
+
+    Body:
+
+    .. code-block:: python
+
+        {
+            "vocabulary_name": string
+        }
+
+
+    Returns:
+        A list of vocabulary terms.
+
+        .. code-block:: python
+
+            {
+                "vocabulary": [{
+                    "Phrase": string,
+                    "IPA": string,
+                    "SoundsLike": string,
+                    "DisplayAs": string
+                    },
+                    ...
+            }
+
+    Raises:
+        500: ChaliceViewError - internal server error
+    """
+    print('download_vocabulary request: '+app.current_request.raw_body.decode())
+    transcribe_client = boto3.client('transcribe', region_name=os.environ['AWS_REGION'])
+    vocabulary_name = json.loads(app.current_request.raw_body.decode())['vocabulary_name']
+    url = transcribe_client.get_vocabulary(VocabularyName=vocabulary_name)['DownloadUri']
+    import urllib.request
+    vocabulary_file = urllib.request.urlopen(url).read().decode("utf-8")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *urlopen*. The resource is allocated by call *request.urlopen*. Execution paths that do not contain closure statements were detected. To prevent this resource leak, close the object returned by *urlopen()* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).